src/sentry/auth/providers/saml2.py

```diff
@@ -67,6 +77,7 @@ def dispatch(self, request, helper):
         # not have been configured yet, build the config first from the state
         if not provider.config:
             provider.config = provider.build_config(helper.fetch_state())
+            disable_2fa_required(request, helper.organization)
```

Comment:
Could you update the comment above this if to read more along the lines of 'During the setup pipeline these two things need to happen'